### PR TITLE
virtqueue cleanup

### PR DESCRIPTION
**Pull request by galak**
Some cleanups to the virtqueue implementation to remove some dead code and reduce the struct virtqueue size to drastically reduce the memory requirements for dynamic allocation.

### CODE REVIEW

File: lib/virtio/virtqueue.c [hunk 1]
```diff
@@ -81,10 +81,9 @@ int virtqueue_create(struct virtio_device *virt_dev, unsigned short id,
 		vq->shm_io = shm_io;
 
 		//TODO : Whether we want to support indirect addition or not.
```
Comment (wjliang) [inline lib/virtio/virtqueue.c]: can also remove this comments

File: lib/virtio/virtqueue.c [hunk 2]
```diff
@@ -303,11 +302,8 @@ void virtqueue_free(struct virtqueue *vq)
 				  "%s: freeing non-empty virtqueue\r\n",
 				  vq->vq_name);
 		}
-		//TODO : Need to free indirect buffers here
 
-		if (vq->vq_ring_mem != NULL) {
-			vq->vq_ring_mem = NULL;
-		}
+		//TODO : Need to free indirect buffers here
```
Comment (wjliang) [inline lib/virtio/virtqueue.c]: we can also remove this comments

Comment (galak) [reply lib/virtio/virtqueue.c]: Removed

File: lib/virtio/virtqueue.c (hunk 1 above)
Comment (galak) [reply lib/virtio/virtqueue.c]: Removed comment

**Comment by galak** [comment]
Removed TODO comments in virtqueue completely, any such items we can track with git hub issues going forward.  I opened Issue #54 for indirect descriptor support.

**Pull request merged**